src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/ExternalMovementsSyncService.kt

```diff
@@ -130,6 +132,9 @@ class ExternalMovementsSyncService(
       }
   }
 
+  private suspend fun requireParentApplicationExists(nomisApplicationId: Long) = mappingApiService.getApplicationMapping(nomisApplicationId)
+    ?: throw ParentEntityNotFoundRetry("Application $nomisApplicationId not created yet so children cannot be processed")
+
```

Comment:
This is where we fail if the parent application has not been created yet. The exception should be caught, its message added to telemetry, and then the exception re-thrown to put the message onto the DLQ. The error handling is currently done in the `track` helper function.